Support/lib/executable.rb

```diff
@@ -0,0 +1,106 @@
+    #    been installed for the current Ruby version.)
+    #
+    # Both RVM and rbenv are supported, too:
+    #
+    # * If RVM is installed and the current directory contains an RVM project
+    #   file, the command to run the executable will be prefixed with `rvm .
+    #   do`. (Note that this is NOT the case if an appropriate `TM_*`
+    #   environment variable is present: In this case, the value of the
+    #   environment variable is returned unchanged. )
+    #
+    # * rbenv just works out of the box as long as your PATH (inside TextMate)
+    #   is setup to contain `~/.rbenv/shims`.
+    #
+    def find(name, env_var = nil)
+      # Safeguard against invalid names so that we don’t need to care about
+      # shell escaping later on
+      raise ArgumentError, "Invalid characters found in '#{name}'" unless name =~ /\A[\w_-]+\z/
+
+      env_var ||= 'TM_' + name.gsub(/\W+/, '_').upcase
+      prefix = determine_rvm_prefix || []
+      if (cmd = ENV[env_var]) && cmd != ''
+        cmd = cmd.shellsplit
+        if system('which', '-s', cmd[0])
+          cmd
+        else
+          raise NotFound, "#{env_var} is set to '#{cmd}', but this does not seem to exist."
+        end
+
+      elsif File.exist?("bin/#{name}")
+        prefix + ["bin/#{name}"]
+
+      elsif File.exist?('Gemfile.lock') && File.read('Gemfile.lock') =~ /^    #{name} /
+        prefix + %W(bundle exec #{name})
+
+      elsif (path = `which #{name}`.chomp) != ''
+        # rbenv installs shims that are present even if the command has not been
+        # installed for the current Ruby version, so we need to also check `rbenv
+        # which` in this case.
+        if path.include?('rbenv/shims') && !system("rbenv which #{name} &>/dev/null")
+          raise NotFound, "rbenv reports that '#{name}' is not installed for the current Ruby version."
+        else
+          prefix + [name]
+        end
+
+      else
+        raise NotFound, "Could not find executable '#{name}'."
+      end
+    end
+
+    private
+
+    # Return appropriate prefix for running commands via RVM if RVM is installed
+    # and the current directory contains an RVM project file.
+    def determine_rvm_prefix
+      rvm = "#{ENV['HOME']}/.rvm/bin/rvm"
+      %W(#{rvm} . do) if File.exist?(rvm) && (rvm_project_file_present? || ruby_version_in_gemfile?)
+    end
+
+    def rvm_project_file_present?
+      # It would be nice if we wouldn’t need to hardcode this list of
+      # filenames here, but so far I couldn’t find any other solution. Using
+      # `rvm current` does not work unfortunately, see
+      # https://github.com/textmate/ruby.tmbundle/pull/104#r78135377
+      project_files = %w(.rvmrc .versions.conf .ruby-version .rbfu-version .rbenv-version)
+      project_files.any?{ |f| File.exist?(f) }
+    end
+
+    def ruby_version_in_gemfile?
+      File.exist?('Gemfile.lock') && File.read('Gemfile.lock') =~ /^RUBY_VERSION/
```

Comment:
This might be an edge case, but if one implements a command to run `bundle install` and uses this method to find `bundle` the lock file might not exist yet.